cumulus_etl/fhir/fhir_utils.py

```diff
@@ -244,36 +244,46 @@ async def _get_docref_note_from_attachment(client: "FhirClient", attachment: dic
     raise ValueError("No data or url field present")
 
 
-def _get_cached_docref_note_path(docref: dict) -> str:
-    return f"{common.get_temp_dir('notes')}/{docref['id']}.txt"
+def _get_cached_note_path(resource: dict) -> str:
+    return f"{common.get_temp_dir('notes')}/{resource['resourceType']}.{resource['id']}.txt"
 
 
-def _get_cached_docref_note(docref: dict) -> str | None:
-    note_path = _get_cached_docref_note_path(docref)
+def _get_cached_note(resource: dict) -> str | None:
+    note_path = _get_cached_note_path(resource)
     try:
         return common.read_text(note_path)
     except FileNotFoundError:
         return None
 
 
-def _save_cached_docref_note(docref: dict, note: str) -> None:
-    note_path = _get_cached_docref_note_path(docref)
+def _save_cached_note(resource: dict, note: str) -> None:
+    note_path = _get_cached_note_path(resource)
     common.write_text(note_path, note)
 
 
-async def get_docref_note(client: "FhirClient", docref: dict) -> str:
+async def get_clinical_note(client: "FhirClient", resource: dict) -> str:
     """
-    Returns the clinical note contained in or referenced by the given docref.
+    Returns the clinical note contained in or referenced by the given resource.
 
     It will try to find the simplest version (plain text) or convert html to plain text if needed.
 
     This also caches the note for the duration of the ETL, to avoid redundant downloads.
     """
-    note = _get_cached_docref_note(docref)
+    note = _get_cached_note(resource)
     if note is not None:
         return note
 
-    attachments = [content["attachment"] for content in docref["content"]]
+    match resource["resourceType"]:
+        case "DiagnosticReport":
+            attachments = resource.get("presentedForm", [])
+        case "DocumentReference":
```

Comment:
How would you feel about moving the expected note containing types to an enum or a set or something like that?

Reply:
I'm not opposed to that, but "what you do with that information next" is often unique to the task the code is doing. Like here, in the highlighted code, how would a set variable like `NOTE_TYPES = {"DiagnosticReport", "DocumentReference"}` affect the `match` tree?

Reply:
Unless maybe there's a like a `notes_utils.py` class that has a swiss army knife of utilities - `get_attachments_from_note` or something like that?

Reply:
But actually, I guess that is what the methods in this file sort of are becoming.

Reply:
Well an enum (or set of constant vars) might make sense for resource type names in general, to avoid typos. Hopefully tests catch typos, but why not defense in depth.

I'll have that in the back of my mind for the future / a more explicit set of note-oriented utils. But I am emotionally ready to just land the current set of fixes and not tweak them.

I already have to fix why the tests are failing in CI but not locally :frowning: